src/backend/distributed/metadata/dependency.c

```diff
@@ -1307,9 +1324,18 @@ ExpandCitusSupportedTypes(ObjectAddressCollector *collector, ObjectAddress targe
 			 * create all objects required by the indices before we create the table
 			 * including indices.
 			 */
-
 			List *indexDependencyList = GetRelationIndicesDependencyList(relationId);
 			result = list_concat(result, indexDependencyList);
+
+			/*
+			 * Get the dependencies of the rule for the given relation. PG keeps internal
+			 * dependency between relation and rule. As it is stated on the PG doc, if
+			 * there is an internal dependency, NORMAL and AUTO dependencies of the
+			 * dependent object behave much like they were dependencies of the referenced
+			 * object.
+			 */
+			List *ruleRefDepList = GetRelationRuleReferenceDependencyList(relationId);
```

Comment:
For a rule on a table, that does not show up as internal dependency. Hence, this code is only triggered for views. Lets make it more explicit?